Sources/Public/Animation/LottiePlaybackMode.swift

```diff
@@ -3,115 +3,155 @@
+  @available(*, deprecated, renamed: "LottiePlaybackMode.play(_:)", message: "Will be removed in a future major release.")
   case fromMarker(
     _ fromMarker: String?,
     toMarker: String,
     playEndMarkerFrame: Bool = true,
     loopMode: LottieLoopMode)
 
-  /// Plays the animation from a named marker to the end of the marker's duration.
-  ///
-  /// A marker is a point in time with an associated duration that is encoded into the
-  /// animation data and assigned a name.
-  ///
-  /// NOTE: If marker is not found the play command will exit.
-  ///
-  /// - Parameter marker: The start marker for the animation playback.
-  /// - Parameter loopMode: The loop behavior of the animation.
+  @available(*, deprecated, renamed: "LottiePlaybackMode.play(_:)", message: "Will be removed in a future major release.")
   case marker(_ marker: String, loopMode: LottieLoopMode)
 
-  /// Plays the given markers sequentially in order.
-  ///
-  /// A marker is a point in time with an associated duration that is encoded into the
-  /// animation data and assigned a name. Multiple markers can be played sequentially
-  /// to create programmable animations.
-  ///
-  /// If a marker is not found, it will be skipped.
-  ///
-  /// If a marker doesn't have a duration value, it will play with a duration of 0
-  /// (effectively being skipped).
-  ///
-  /// If another animation is played (by calling any `play` method) while this
-  /// marker sequence is playing, the marker sequence will be cancelled.
-  ///
-  /// - Parameter markers: The list of markers to play sequentially.
+  @available(*, deprecated, renamed: "LottiePlaybackMode.play(_:)", message: "Will be removed in a future major release.")
   case markers(_ markers: [String])
+
+  // MARK: Public
+
+  public enum PausedState: Hashable {
+    /// Any existing animation will be paused at the current frame.
+    case currentFrame
+
+    /// The animation is paused at the given progress value,
+    /// a value between 0.0 (0% progress) and 1.0 (100% progress).
+    case progress(_ progress: AnimationProgressTime)
+
+    /// The animation is paused at the given frame of the animation.
+    case frame(_ frame: AnimationFrameTime)
+
+    /// The animation is paused at the given time value from the start of the animation.
+    case timestamp(_ time: TimeInterval)
+
+    /// Pauses the animation at a given marker and position
+    case marker(_ name: String, position: LottieMarkerPosition = .start)
+  }
+
+  public enum PlaybackMode: Hashable {
+    /// Plays the animation from a progress (0-1) to a progress (0-1).
+    /// - Parameter from: The start progress of the animation. If `nil` the animation will start at the current progress.
+    /// - Parameter to: The end progress of the animation.
+    /// - Parameter loopMode: The loop behavior of the animation.
+    case progress(from: AnimationProgressTime? = nil, to: AnimationProgressTime, loopMode: LottieLoopMode)
```

Comment:
Can we leave the names of these as-is? I prefer `.play(.fromProgress(0, toProgress: 1))` over `.play(.progress(from: 1, to: 2))`. This reads more fluently and also matches the naming of the UIKit APIs on `LottieAnimationView`

Reply:
```suggestion
    case fromProgess(_ fromProgress: AnimationProgressTime? = nil, toProgress: AnimationProgressTime, loopMode: LottieLoopMode)
```